bottle.py

```diff
@@ -2813,13 +2813,15 @@ def abort(code=500, text='Unknown Error.'):
     raise HTTPError(code, text)
 
 
-def redirect(url, code=None):
+def redirect(url, code=None, cache=None):
     """ Aborts execution and causes a 303 or 302 redirect, depending on
         the HTTP protocol version. """
     if not code:
         code = 303 if request.get('SERVER_PROTOCOL') == "HTTP/1.1" else 302
     res = response.copy(cls=HTTPResponse)
     res.status = code
+    if type(cache) is int:
```

Comment:
You can check if the variable exists, you don't need to look at its type.
If it is an unexpected string it will be the developer's responsibility